factory/api/apiResolverFactory.go

```diff
@@ -647,25 +641,27 @@ func createBuiltinFuncs(
 	guardedAccountHandler vmcommon.GuardedAccountHandler,
 	automaticCrawlerAddresses [][]byte,
 	maxNumAddressesInTransferRole uint32,
-	dnsV2Addresses [][]byte,
+	dnsV2Addresses []string,
+	whiteListedCrossChainAddresses []string,
+	selfESDTPrefix []byte,
+	pubKeyConverter core.PubkeyConverter,
 ) (vmcommon.BuiltInFunctionFactory, error) {
-	mapDNSV2Addresses := make(map[string]struct{})
-	for _, address := range dnsV2Addresses {
-		mapDNSV2Addresses[string(address)] = struct{}{}
-	}
-
 	argsBuiltIn := builtInFunctions.ArgsCreateBuiltInFunctionContainer{
-		GasSchedule:               gasScheduleNotifier,
-		MapDNSAddresses:           make(map[string]struct{}),
-		MapDNSV2Addresses:         mapDNSV2Addresses,
-		Marshalizer:               marshalizer,
-		Accounts:                  accnts,
-		ShardCoordinator:          shardCoordinator,
-		EpochNotifier:             epochNotifier,
-		EnableEpochsHandler:       enableEpochsHandler,
-		GuardedAccountHandler:     guardedAccountHandler,
-		AutomaticCrawlerAddresses: automaticCrawlerAddresses,
-		MaxNumNodesInTransferRole: maxNumAddressesInTransferRole,
+		GasSchedule:                    gasScheduleNotifier,
+		MapDNSAddresses:                make(map[string]struct{}),
+		DNSV2Addresses:                 dnsV2Addresses,
+		WhiteListedCrossChainAddresses: whiteListedCrossChainAddresses,
+		EnableUserNameChange:           false,
+		Marshalizer:                    marshalizer,
+		Accounts:                       accnts,
+		ShardCoordinator:               shardCoordinator,
+		EpochNotifier:                  epochNotifier,
+		EnableEpochsHandler:            enableEpochsHandler,
+		GuardedAccountHandler:          guardedAccountHandler,
+		AutomaticCrawlerAddresses:      automaticCrawlerAddresses,
+		MaxNumNodesInTransferRole:      maxNumAddressesInTransferRole,
```

Comment:
MaxNumNodesInTransferRole or maxNumAddressesInTransferRole -> nodes or addresses?

Reply:
Fixed it. Renamed to maxNumAddressesInTransferRole